internal/core/window.rs

```diff
@@ -135,6 +136,13 @@ pub trait PlatformWindow {
     /// Resizes the window to the specified size on the screen, in physical pixels and excluding
     /// a window frame (if present).
     fn set_inner_size(&self, size: euclid::Size2D<u32, PhysicalPx>);
+
+    /// Sends the given text into the system clipboard
+    fn set_clipboard_text(&self, _text: String) {}
```

Comment:
API question:
Should that be String or SharedString?
Or maybe &str?

Reply:
API wise a SharedString makes most sense, this is for consumption. Implementation wise it's a bit silly:

```rust
fn selected_text(self: Pin<&Self>) -> String {
        let (anchor, cursor) = self.selection_anchor_and_cursor();
        let text: String = self.text().into();
        text.split_at(anchor).1.split_at(cursor - anchor).0.into()
}
```

Suppose we had a property getter that gave us a `&SharedString` then we could get a `&str`.

So I suggest this then:

```rust
fn with_selected_text(self: Pin<&Self>, cb: impl FnOnce(&str)) {
        let (anchor, cursor) = self.selection_anchor_and_cursor();
        let text: String = self.text().into();
        cb(text.split_at(anchor).1.split_at(cursor - anchor).0)
}

fn copy(self: Pin<&Self>) {
    self.with_selected_text(|text| clipboard.set_text(text));
}
```

What do you think?

Reply:
I've had the opportunity to stare at `selected_text` durring a few calls, and it hurts my eyes :-p
(the `text().into()` part, and the `split_at()` instead of `[...]`).

Reply:
Is that a yes? ;-)

Reply:
Since `selected_text` is private api, i don't care so much. I think it is ok to allocate anyway.
`set_clipboard_text` and `clipboard_text`  is going to end up public api so we need to think carefully.

The question is how will the backend store the text:
 - the backend based on copy_pasta always need and return a String anyway
 - the Qt backend will need to convert to/from QString
 - The mcu backend could keep either a String, or a SharedString.  SharedString would do less allocations.

Then how do we call this. We could return a SharedString or a String from `selected_text()` so that's the same.  And if we want to be extra efficient and pass a &str from a `with_selected_text` we can also do that.
And for paste, we kind of need just the &str to pass to the insert function.  If the line edit is empty and it was a SharedString we could optimize. But all of that doesn't really matter as we don't do copy/paste in a loop.

So i'd say for now, `String` is good enough and we can review that again when we do the final review of the future WindowBackend trait.

Reply:
Ok. I have a changeset for `&str` and I'll make a separate PR - I found what I think is a compelling argument :)